database/sql/src/main/scala/cromwell/database/slick/MetadataSlickDatabase.scala

```diff
@@ -133,7 +138,7 @@ class MetadataSlickDatabase(originalDatabaseConfig: Config)
         updateCount <- dataAccess.
           customLabelEntriesForWorkflowExecutionUuidAndLabelKey(
             (customLabelEntry.workflowExecutionUuid, customLabelEntry.customLabelKey)
-          ).update(customLabelEntry)
+          ).update(customLabelEntry.toUpdateColumns)
```

Comment:
@kshakir The specific error that motivated this change (and several others) was this:
```
2019-06-17 15:07:00,298 cromwell-system-akka.dispatchers.service-dispatcher-94 ERROR - Error summarizing metadata
org.postgresql.util.PSQLException: ERROR: null value in column "CUSTOM_LABEL_ENTRY_ID" violates not-null constraint
  Detail: Failing row contains (null, cromwell-workflow-id, cromwell-a8afcb27-4411-4c83-9b38-8402a23cc579, a8afcb27-4411-4c83-9b38-8402a23cc579).
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
        at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441)
        at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365)
        at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:143)
        at org.postgresql.jdbc.PgPreparedStatement.executeUpdate(PgPreparedStatement.java:120)
        at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
        at slick.jdbc.JdbcActionComponent$UpdateActionExtensionMethodsImpl$$anon$8.$anonfun$run$10(JdbcActionComponent.scala:325)
        at slick.jdbc.JdbcActionComponent$UpdateActionExtensionMethodsImpl$$anon$8.$anonfun$run$10$adapted(JdbcActionComponent.scala:321)
        at slick.jdbc.JdbcBackend$SessionDef.withPreparedStatement(JdbcBackend.scala:386)
        at slick.jdbc.JdbcBackend$SessionDef.withPreparedStatement$(JdbcBackend.scala:381)
        at slick.jdbc.JdbcBackend$BaseSession.withPreparedStatement(JdbcBackend.scala:448)
        at slick.jdbc.JdbcActionComponent$UpdateActionExtensionMethodsImpl$$anon$8.run(JdbcActionComponent.scala:321)
        at slick.jdbc.JdbcActionComponent$UpdateActionExtensionMethodsImpl$$anon$8.run(JdbcActionComponent.scala:320)
        at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:30)
        at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:27)
        at slick.basic.BasicBackend$DatabaseDef$$anon$2.liftedTree1$1(BasicBackend.scala:275)
        at slick.basic.BasicBackend$DatabaseDef$$anon$2.run(BasicBackend.scala:275)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```
(This is also what prevents us from doing "upserts" in Postgres.)

Reply:
Also:
```
cromwell=# \d "CUSTOM_LABEL_ENTRY";
                                                    Table "public.CUSTOM_LABEL_ENTRY"
         Column          |          Type          |                                      Modifiers
-------------------------+------------------------+--------------------------------------------------------------------------------------
 CUSTOM_LABEL_ENTRY_ID   | bigint                 | not null default nextval('"CUSTOM_LABEL_ENTRY_CUSTOM_LABEL_ENTRY_ID_seq"'::regclass)
 CUSTOM_LABEL_KEY        | character varying(255) | not null
 CUSTOM_LABEL_VALUE      | character varying(255) | not null
 WORKFLOW_EXECUTION_UUID | character varying(100) | not null
Indexes:
    "PK_CUSTOM_LABEL_ENTRY" PRIMARY KEY, btree ("CUSTOM_LABEL_ENTRY_ID")
    "UC_CUSTOM_LABEL_ENTRY_CLK_CLV_WEU" UNIQUE CONSTRAINT, btree ("CUSTOM_LABEL_KEY", "CUSTOM_LABEL_VALUE", "WORKFLOW_EXECUTION_UUID")
    "IX_CUSTOM_LABEL_ENTRY_CLK_CLV" btree ("CUSTOM_LABEL_KEY", "CUSTOM_LABEL_VALUE")
    "SYS_IDX_11226" btree ("WORKFLOW_EXECUTION_UUID")
Foreign-key constraints:
    "FK_CUSTOM_LABEL_ENTRY_WORKFLOW_EXECUTION_UUID" FOREIGN KEY ("WORKFLOW_EXECUTION_UUID") REFERENCES "WORKFLOW_METADATA_SUMMARY_ENTRY"("WORKFLOW_EXECUTION_UUID") ON DELETE CASCADE```

Reply:
Re PostgreSQL+Slick/Liquibase:

The other DSP-Batch'ers and I discussed changing the old liquibase changelogs, modifying them to be able to run on both PostgreSQL+MySQL+HSQLDB. During the discussion I was convinced to go another way than what I already told you.

I left a few comments over in this example commit: https://github.com/broadinstitute/cromwell/commit/77630aaa90cb27af9c5d9360a5e8a2c0660a1e10

(For those who come across this message in the future, the commit may disappear but the gist of the changes should make it into this PR.)

The commit goes about explicitly-including supported databases. What's preferred with this approach is:
1. No changes to the old MD5s
2. If needed we can do the same thing for additional DBMS (ex: MariaDB)

Using liquibase's changelog generator (plus some minor tweaks) I've partially recreated and attached the schemas from MySQL. Hopefully the XML files are a good starting point for PostgreSQL.

[engine_schema.xml](https://github.com/broadinstitute/cromwell/files/3308568/engine_schema.xml.txt)
[metadata_schema.xml](https://github.com/broadinstitute/cromwell/files/3308569/metadata_schema.xml.txt)

Later, towards the end of the commit, hopefully the examples in the package `cromwell.database` will workaround the issues with AutoInc and Lob columns.

Reply:
@kshakir Thanks, I'll take a look at 77630aa.  Should I try incorporating this into my PR?

Reply:
Yes please, do try replicating the example changes across your PR. Thanks again for your patience with all of this.

Reply:
I've just started on the migrations, but the Scala code changes should be complete now if you want to start reviewing them.  So far everything is working perfectly for me in my local environment.

Reply:
The migrations are now done too.  I find this much cleaner than what I had before.